src/transformer.ts

```diff
@@ -200,7 +200,12 @@ function Transformer(context: ts.TransformationContext) {
     }
     for (const dec of node.decorators) {
       if (dec.kind == ts.SyntaxKind.Decorator) {
-        if (transformConfig.decoratorNames.indexOf(dec.expression.getText()) != -1) {
+        // Find decorator name. If decorator is called as function, then name is at first child.
+        const dec_name = (dec.expression.getChildCount() === 0) ?
```

Comment:
See the example below, but the issue was that previously the dec.expression.getText() call would return:

```
@ParamDecorator({
 +    strParam: 'string_value',
 +})
```
Which would not match the name of the decorator `ParamDecorator`

This change takes that type of usage into account by looking down the AST a bit further in the case of a called decorator.